powerform/pf-deployment-context/classes/pf-deployment-context.ps1

```diff
@@ -1,4 +1,12 @@
 Class PfDeploymentContext{
     [System.Collections.ArrayList]$Resources=@()
+
     PfDeploymentContext(){}
+
+    [void]Deploy(){
+        Foreach($resource in $this.Resources){
+            $access = (New-Object -TypeName "$($resource.AzResourceType.ClassPrefix)Access" -ArgumentList $resource)
```

Comment:
Moved instantiating the access object within the deployment context.  I did not like having the access object hang off the buildstate and never used.  Also, I now pass in the buildstate to the access constructor.  This now alleviates having to pass "buildstate" to every method in the access class.  I know you wanted to look into maybe using functions.  This is something I haven't looked into yet.

Reply:
yeah, this good.  i think would seem replaceable via a dynamically invoked function, but this makes state back into a parm on all 'child' functions again.   i think though from testability perspective that going to probably be path we need to go

Reply:
This is now a private function